cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -362,3 +384,144 @@ func (m *McmManager) GetMachineDeploymentNodes(machinedeployment *MachineDeploym
 	}
 	return nodes, nil
 }
+
+//GetMachineDeploymentNodeTemplate returns the NodeTemplate which belongs to the MachineDeployment.
+func (m *McmManager) GetMachineDeploymentNodeTemplate(machinedeployment *MachineDeployment) (*nodeTemplate, error) {
+
+	md, err := m.machineclient.MachineDeployments(m.namespace).Get(machinedeployment.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("Unable to fetch MachineDeployment object %s, Error: %v", machinedeployment.Name, err)
+	}
+
+	var region string
+	var tags map[string]string
+	var instance instanceType
+	machineClass := md.Spec.Template.Spec.Class
+	switch machineClass.Kind {
+	case "AWSMachineClass":
+		mc, err := m.machineclient.AWSMachineClasses(m.namespace).Get(machineClass.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("Unable to fetch AWSMachineClass object %s, Error: %v", machinedeployment.Name, err)
+		}
+		awsInstance := aws.InstanceTypes[mc.Spec.MachineType]
+		instance = instanceType{
+			InstanceType: awsInstance.InstanceType,
+			VCPU:         awsInstance.VCPU,
+			MemoryMb:     awsInstance.MemoryMb,
+			GPU:          awsInstance.GPU,
+		}
+		region = mc.Spec.Region
+		tags = mc.Spec.Tags
+	case "AzureMachineClass":
+		mc, err := m.machineclient.AzureMachineClasses(m.namespace).Get(machineClass.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("Unable to fetch AzureMachineClass object %s, Error: %v", machinedeployment.Name, err)
+		}
+		azureInstance := azure.InstanceTypes[mc.Spec.Properties.HardwareProfile.VMSize]
+		instance = instanceType{
+			InstanceType: azureInstance.InstanceType,
+			VCPU:         azureInstance.VCPU,
+			MemoryMb:     azureInstance.MemoryMb,
+			GPU:          azureInstance.GPU,
+		}
+		region = mc.Spec.Location
+		tags = mc.Spec.Tags
+	default:
+		return nil, cloudprovider.ErrNotImplemented
+	}
+
+	nodeTmpl := &nodeTemplate{
+		InstanceType: &instance,
+		Region:       region,
+		Zone:         "undefined",
```

Comment:
Zone is not yet defined on the machine class, we should probably add it.

Reply:
Yes, we should probably do that. 
Btw do you know, what could be the probable implications on Autoscaler's behavior, for having an "undefined" zone here?

Reply:
Region, InstanceType name are set as labels through
https://github.com/gardener/autoscaler/pull/26/files#diff-50934888f33c34f8e25393b844c0c8c6R485

Then the autoscaler use the kubernetes scheduler to check if a pod is allocable into a node part of the machinedeployment. see https://github.com/gardener/autoscaler/blob/machine-controller-manager/cluster-autoscaler/simulator/predicates.go#L250 

So if the zone is not defined and you have an affinity rule on your Pod Deployment allowing to run the pod only on Zone X, then the pod won't be schedulable when the machinedeployment is set to 0 which mean the node template is being used, if the number of replicas is greater than 0 the austocaler will describe a existing node part of the cluster and the zone will be defined therefore the pod will match the taints/labels and the Zone label. 

It would be good to add Zone/DefaultOS/DefaultArch as part of the machineClass.